### ISSUE TITLE
Add a mechanism to validate that unit tests are implemented in the`tests` mod block.

### ISSUE THREAD

**Issue by sandreim** [firecracker-microvm/firecracker#1704]
Kcov reports coverage only for tests defined in the `mod tests` block, so we need to enforce this to avoid bogus coverage results.
https://github.com/firecracker-microvm/firecracker/blob/e6436c599756b653931a57d5f5c418f7d22cac40/tests/integration_tests/build/test_coverage.py#L61

**Comment by andreeaflorescu** [comment]
Just a tiny clarification: kcov reports the coverage for all the tests, but the problem is that the report that calculates the coverage is wrong without having the test section excluded because it is counting the test line as code lines as well.

**Comment by tnotw** [comment]
Would this work?  An additional test in the test_coverage module that diffs the list of files containing the "#[cfg(test)]" attribute with the list of files containing the string "mod tests {".   
```
+def test_ensure_mod_tests():
+    """Fail if mod_tests hasn't been defined as the test module name."""
+    # List all source files containing rust test attribute.
+    # List all files containing string "mod tests {".
+    # Diff the two lists.
+    # Diff returns zero if file lists match, non-zero if different.
+    # Redirect file diff list to stderr.
+    cmd = (
+        '/bin/bash '
+        '-c '
+        '"diff '
+        '<(grep --extended-regexp '
+        '--files-with-matches '
+        '--recursive '
+        '\'^#\\[cfg\\(test\\)\\]$\' ../src) '
+        '<(grep '
+        '--files-with-matches '
+        '--recursive '
+        '\'mod tests {\' ../src)" 1>&2'
+    )
```
Giving: 
```[ unsafe | 1 tests | 1 worker(s) ]
  FAILED    integration_tests/build/test_coverage.py::test_ensure_mod_tests

========================================================================================== FAILURES ===========================================================================================
____________________________________________________________________________________ test_ensure_mod_tests ____________________________________________________________________________________
integration_tests/build/test_coverage.py:67: in test_ensure_mod_tests
    stdout=PIPE
/usr/lib/python3.6/subprocess.py:438: in run
    output=stdout, stderr=stderr)
E   subprocess.CalledProcessError: Command '/bin/bash -c "diff <(grep --extended-regexp --files-with-matches --recursive '^#\[cfg\(test\)\]$' ../src) <(grep --files-with-matches --recursive 'mod tests {' ../src)" 1>&2' returned non-zero exit status 1.
------------------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------------------
26,29d25
< ../src/cpuid/src/transformer/amd.rs
< ../src/cpuid/src/transformer/common.rs
< ../src/cpuid/src/transformer/intel.rs
< ../src/cpuid/src/transformer/mod.rs
```
It might be a bit leaky,  but if this is essentially just to catch "mod test" typos, would it be sufficient?

**Comment by lauralt** [comment]
Fixed by #1833.
